db/query.go

```diff
@@ -129,30 +197,60 @@ const (
 	QueryParamOlderThan   = "olderThan"
 )
 
+// N1QlQueryWithStats is a wrapper for gocbBucket.Query that performs additional diagnostic processing (expvars, slow query logging)
+func (context *DatabaseContext) N1QLQueryWithStats(queryName string, statement string, params interface{}, consistency gocb.ConsistencyMode, adhoc bool) (results gocb.QueryResults, err error) {
+
+	if base.SlowQueryWarningThreshold > 0 {
+		defer base.SlowQueryLog(time.Now(), "N1QL Query(%q)", queryName)
+	}
+
+	gocbBucket, ok := base.AsGoCBBucket(context.Bucket)
+	if !ok {
+		return nil, errors.New("Cannot perform principals N1QL query on non-Couchbase bucket.")
```

Comment:
I think the word "principals" doesn't belong here.

Reply:
Thanks - fixed.